lib/resources/resourceBase.js

```diff
@@ -66,6 +66,10 @@ var ResourceBase = function (endpoint, config) {
           return reject(body.error);
         }
 
+        if (resp && resp.statusCode >= 500) {
+          return reject(resp);
```

Comment:
`resp` is a huge object and I don't think you want to reject with the whole thing. You should instead reject with something of type Error and with a nice string message, if you can pull one out.